src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -870,28 +913,51 @@ function transitionHeatmapTrack(
                 : undefined
             )
         };
-        const newTrackId = oncoprint.addTracks([heatmapTrackParams])[0];
+        // register new track in oncoprint
+        const newTrackId:number = oncoprint.addTracks([heatmapTrackParams])[0];
+        // store relation between React heatmap track specs and OncoprintJS trackIds
         trackSpecKeyToTrackId[nextSpec.key] = newTrackId;
 
-        let trackIdForRuleSetSharingKey:"heatmap"|"heatmap01" = "heatmap";
-        if (nextSpec.molecularAlterationType === "METHYLATION") {
-            trackIdForRuleSetSharingKey = "heatmap01";
-        }
-        if (typeof trackIdForRuleSetSharing[trackIdForRuleSetSharingKey] !== "undefined") {
-            oncoprint.shareRuleSet(trackIdForRuleSetSharing[trackIdForRuleSetSharingKey]!, newTrackId);
+        if (nextSpec.molecularAlterationType !== AlterationTypeConstants.TREATMENT_RESPONSE) {
+            let rulesetTrackIdKey:"heatmap"|"heatmap01";
+            if (nextSpec.molecularAlterationType !== AlterationTypeConstants.METHYLATION) {
+                rulesetTrackIdKey = "heatmap01";
+            } else {
+                rulesetTrackIdKey = "heatmap";
+            }
+            // if the track is a molecular profile, check whether there is an existing ruleset that applies to this track
+            if (trackIdForRuleSetSharing[rulesetTrackIdKey] !== undefined) {
+                // if so, associate the is of the new track so that its is formatted by the existing ruleset
+                oncoprint.shareRuleSet(trackIdForRuleSetSharing[rulesetTrackIdKey]!, newTrackId);
+            } else {
+                // if not, make this track available to other new tracks for ruleset sharing
+                trackIdForRuleSetSharing[rulesetTrackIdKey] = newTrackId;
```

Comment:
there is no `else` in the original logic, have you added it here on purpose? this line should happen unconditionally

Reply:
Hmm, the code in the most recent version is as this:
```
        if (nextSpec.molecularAlterationType !== AlterationTypeConstants.TREATMENT_RESPONSE) {
            let rulesetTrackIdKey:"heatmap"|"heatmap01";
            if (nextSpec.molecularAlterationType === AlterationTypeConstants.METHYLATION) {
                rulesetTrackIdKey = "heatmap01";
            } else {
                rulesetTrackIdKey = "heatmap";
            }
            // if the track is a molecular profile, check whether there is an existing ruleset that applies to this track
            if (trackIdForRuleSetSharing[rulesetTrackIdKey] !== undefined) {
                // if so, associate the is of the new track so that its is formatted by the existing ruleset
                oncoprint.shareRuleSet(trackIdForRuleSetSharing[rulesetTrackIdKey]!, newTrackId);
            } else {
                // if not, make this track available to other new tracks for ruleset sharing
                trackIdForRuleSetSharing[rulesetTrackIdKey] = newTrackId;
            }
        } else {
            // if the track is a treatment profile, always add to trackIdForRuleSetSharing
            // this makes the trackId available for existing tracks fir ruleset sharing
            trackIdForRuleSetSharing.treatment![nextSpec.molecularProfileId] = newTrackId;
        }
```

You can see that this `else` is for treatment tracks. I did add a comment for this to clarify this point.

Reply:
In `rc`, `trackIdForRuleSetSharing[trackIdForRuleSetSharingKey] = newTrackId` runs unconditionally, as I indicated in my original comment.

Reply:
Ok, I think I forgot to fetch. My change is now updated to reflect current `rc`. Moreover, this logic is now covered by `DeltaUtil.spec.ts`.